### PR TITLE
feat: Get All Categories List to Category Remote 

### PR DESCRIPTION
**Pull request by RaniaZeid**

### CODE REVIEW

File: Networking/Sources/Networking/Remote/AllCategoryRemote.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/// Protocol for `AllCategoryRemote` mainly used for mocking.
+///
+public protocol AllCategoryRemoteProtocol {
+    func loadAllCategory(completion: @escaping (Result<[String], Error>) -> Void)
+}
+
+/// Products: Remote Endpoints
+///
+public class AllCategoryRemote: Remote, AllCategoryRemoteProtocol {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Remote/AllCategoryRemote.swift]:
```suggestion
public protocol AllCategoryRemoteProtocol {
    func loadAllCategory(completion: @escaping (Result<[String], Error>) -> Void)
}

/// Products: Remote Endpoints
///
public class CategoryRemote: Remote, CategoryRemoteProtocol {
```

**Review by ahmdmhasn — COMMENTED** [review]
Don't forget to rename the file to `CategoryRemote` since it should include all category related APIs.

**Comment by ahmdmhasn** [comment]
Make sure to make the branch name in all lower case letters to avoid any unexpected errors when working with a team. 
~~feat/get-All-Categories~~ feat/get-all-categories

File: Networking/Sources/Networking/Remote/AllCategoryRemote.swift (hunk 1 above)
Comment (RaniaZeid) [reply Networking/Sources/Networking/Remote/AllCategoryRemote.swift]: ok, I will update it

**Review by ahmdmhasn — APPROVED** [review]

**Pull request merged**